install.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+# The script requires root permissions
+
+set -e
+
+if [ "$EUID" -ne 0 ]; then
+    echo "Please run as root."
+    exit
+fi
+
+function print_help() {
+    cat << END
+This is a trento-agent installer. Trento is a web-based graphical user interface
+that can help you deploy, provision and operate infrastructure for SAP Applications
+
+Usage:
+
+  sudo ./install.sh --agent-bind-ip <192.168.122.10> --server-ip <192.168.122.5>
+
+Arguments:
+  --agent-bind-ip   The private address to which the trento-agent should be bound for internal communications.
+                    This is an IP address that should be reachable by the other nodes, including the trento server.
+  --server-ip       The trento server ip.
+  --help            Print this help.
+END
+}
+
+case "$1" in
+--help)
+    print_help
+    exit 0
+    ;;
+esac
+
+ARGUMENT_LIST=(
+    "agent-bind-ip"
+    "server-ip"
+)
+
+opts=$(
+    getopt \
+        --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}")" \
+        --name "$(basename "$0")" \
+        --options "" \
+        -- "$@"
+)
+
+eval set "--$opts"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+    --agent-bind-ip)
+        AGENT_BIND_IP=$2
+        shift 2
+        ;;
+
+    --server-ip)
+        SERVER_IP=$2
+        shift 2
+        ;;
+
+    *)
+        break
+        ;;
+    esac
+done
+
+if [ -z "$AGENT_BIND_IP" ]; then
+    read -rp "Please provide a bind IP for the agent: " AGENT_BIND_IP < /dev/tty
+fi
+if [ -z "$SERVER_IP" ]; then
+    read -rp "Please provide the server IP: " SERVER_IP < /dev/tty
+fi
+if [ -z "$NODE_NAME" ]; then
+    NODE_NAME="$HOSTNAME"
+fi
+
+TRENTO_REPO_KEY=${TRENTO_REPO_KEY:-"https://download.opensuse.org/repositories/devel:/sap:/trento/15.3/repodata/repomd.xml.key"}
+TRENTO_REPO=${TRENTO_REPO:-"https://download.opensuse.org/repositories/devel:/sap:/trento/15.3/devel:sap:trento.repo"}
+
+CONSUL_VERSION=1.9.6
+CONSUL_PATH=/srv/consul
+CONFIG_PATH="$CONSUL_PATH/consul.d"
+CONSUL_HCL_TEMPLATE='data_dir = "/srv/consul/data/"
+log_level = "DEBUG"
+datacenter = "dc1"
+ui = true
+bind_addr = "@BIND_ADDR@"
+client_addr = "0.0.0.0"
+retry_join = ["@JOIN_ADDR@"]'
+
+CONSUL_SERVICE_NAME="consul.service"
+CONSUL_SERVICE_TEMPLATE='[Unit]
+Description="HashiCorp Consul - A service mesh solution"
+Documentation=https://www.consul.io/
+Requires=network-online.target
+After=network-online.target
+ConditionFileNotEmpty=/srv/consul/consul.d/consul.hcl
+PartOf=trento-agent.service
+
+[Service]
+ExecStart=/srv/consul/consul agent -config-dir=/srv/consul/consul.d
+ExecReload=/bin/kill --signal HUP $MAINPID
+KillMode=process
+Restart=on-failure
+RestartSec=5
+Type=notify
+
+
+[Install]
+WantedBy=multi-user.target'
+
+. /etc/os-release
+if [[ ! $PRETTY_NAME =~ "SUSE" ]]; then
+    echo "Operating system is not supported. Exiting."
+    exit 1
+fi
```

Comment:
Since we are seeing some installation issues even within the SLE15, we should only allow SLE15 installs and not just anything SUSE.

Reply:
Taking granted that the problems with SLE15 were caused by the RPM systemd unit (see comments above) and now fixed, we use the script to install trento also in opensuse and leap for development purposes. 
Do we really want to limit the version to SLE15?

Reply:
It's not required to use SLE: technically, we just need a SUSE distro because the scripts assumes `zypper` is available.